File: backend/app/api/routes/verification.py
from fastapi import APIRouter, HTTPException
from app.services.twilio_service import TwilioService

router = APIRouter(prefix="/verification", tags=["verification"])
twilio_service = TwilioService()

@router.post("/send-code")
async def send_code(phone_number: str):
    result = twilio_service.send_verification_code(phone_number)
    if not result["success"]:
        raise HTTPException(status_code=500, detail=result["error"])
    return {"message": "Code sent successfully"}

@router.post("/verify-code")
async def verify_code(phone_number: str, code: str):
    result = twilio_service.verify_code(phone_number, code)
    if not result["success"]:
        raise HTTPException(status_code=400, detail=result.get("message", result["error"]))
    return {"message": "Code verified successfully"}

File: frontend/src/api-calls/verification.ts
import axios from 'axios';

export const sendVerificationCode = async (phone_number) => {
  try {
    console.log("Sending data to backend:", { phone_number });
    const response = await axios.post(
      'http://localhost:8000/verification/send-code', {
      phone_number
    });

    if (!response.data.message) {
      throw new Error(response.data.detail);
    }

    return response.data;
  } catch (error) {
    console.error("Failed to send code:", error);
    throw error;
  }
};

export const verifyCode = async (phone_number, code) => {
  try {
    const response = await axios.post(
        "http://localhost:8000/verification/verify-code", {
      phone_number,
      code
    });

    if (!response.data.message) {
      throw new Error(response.data.detail);
    }

    return response.data;
  } catch (error) {
    console.error("Verification failed:", error);
    throw error;
  }
};
